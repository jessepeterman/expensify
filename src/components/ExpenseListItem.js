import React from 'react';
import {Link} from 'react-router-dom';

// pass in the prop.expense and destructure the object to be used inside expenseListItem
const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <p>Amount: ${amount}  Created on: {createdAt}</p>
  </div>
);


export default ExpenseListItem;
