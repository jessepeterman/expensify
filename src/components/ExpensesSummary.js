import React from 'react';
import {connect} from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
  const itemWord = props.expenses.length > 1 || props.expenses.length < 1 ? 'items' : 'item';
  const formattedExpensesTotal = numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00');

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Your total for <span>{props.expenses.length}</span> {itemWord} is: <span>{formattedExpensesTotal}</span></h1>
          <div className="page-header__actions">
            <Link className="button" to="/create">Add Expense</Link>
          </div>
      </div>
    </div>
  );
}

// numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
}

export default connect(mapStateToProps)(ExpensesSummary); 