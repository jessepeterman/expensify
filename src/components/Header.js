import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <li><NavLink to="/" activeClassName="is-active" exact>Go Home</NavLink></li>
    <li><NavLink to="/create" activeClassName="is-active" exact>Create Expense</NavLink></li>
    <li><NavLink to="/help" activeClassName="is-active" exact>Help</NavLink></li>
  </header>
);

export default Header;