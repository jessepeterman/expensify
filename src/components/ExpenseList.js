import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from "../selectors/expenses";

const ExpenseList = (props) => (
  <div>
      <h1>Expense List</h1>
    {
      props.expenses.map((expense) => (
      
      <ExpenseListItem 
        key={expense.id}
        // send entire expense from the map, to the expenselistitem component prop argument
        {...expense}
        />
    ))
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
}

export default connect(mapStateToProps)(ExpenseList);