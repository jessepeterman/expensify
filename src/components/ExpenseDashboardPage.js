import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import getExpensesTotal from '../selectors/expenses-total';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;