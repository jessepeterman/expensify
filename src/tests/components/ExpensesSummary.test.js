import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

const expenseList = [{
  id: '1',
  description: 'gum',
  note: '',
  amount: 9434,
  createdAt: 0
}, {
  id: '2',
  description: 'other half',
  note: '',
  amount: 4717,
  createdAt: 100
  }, {
    id: '3',
    description: 'Half',
    note: '',
    amount: 4717,
    createdAt: 100
  }];

test('should render expense summary with', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenseList} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render expense summary with', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenseList[0]]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render expense summary with', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenseList[1], expenseList[2]]}/>);
  expect(wrapper).toMatchSnapshot();
});