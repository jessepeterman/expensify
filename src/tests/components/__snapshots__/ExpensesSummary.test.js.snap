// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary with 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Your total for 
      <span>
        3
      </span>
       
      items
       is: 
      <span>
        $188.68
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render expense summary with 2`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Your total for 
      <span>
        1
      </span>
       
      item
       is: 
      <span>
        $94.34
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render expense summary with 3`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Your total for 
      <span>
        2
      </span>
       
      items
       is: 
      <span>
        $94.34
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
